libavfilter/vf_drawbox.c

```diff
@@ -140,11 +143,29 @@ static void draw_region(AVFrame *frame, DrawBoxContext *ctx, int left, int top,
     }
 }
 
+static enum AVFrameSideDataType box_source_string_parse(const char *box_source_string)
+{
+    av_assert0(box_source_string);
+    if (!strcmp(box_source_string, "side_data_detection_bbox")) {
+        return AV_FRAME_DATA_DETECTION_BBOXES;
+    } else {
+        return -1;
+    }
+}
+
 static av_cold int init(AVFilterContext *ctx)
 {
     DrawBoxContext *s = ctx->priv;
     uint8_t rgba_color[4];
 
+    if (s->box_source_string) {
+        s->box_source = box_source_string_parse(s->box_source_string);
+        if (s->box_source == -1) {
```

Comment:
the enum type cannot be compared with 0, so i used `== -1` to verify the if the retval correct or not

Reply:
i did not use error like `AVERROR(EINVAL)`, because the AVERROR returns value above 0, wihch may be ambiguous with `enum AVFrameSideDataType` value

Reply:
lost the context, but AVERROR(EINVAL) is less than 0.

Reply:
Oh, yes, the AVERROR(EINVAL) is negtative number, 
and what do you mean 'lost the context'?

Reply:
forgot the original comment on this discussion.

Reply:
I changed the retval into `AVERROR(EINVAL)` , but I'm not sure if the function returns  `AVERROR(EINVAL)` with return type enum is OK. I did not find similar code.